domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3097,15 +3197,15 @@ public void select(AsyncResult<SQLConnection> conn, String sql, Tuple params,
   }
 
   /**
-   * Run a select query and return the first record, or null if there is no result.
+   * Run a select query with the readonly connection and return the first record, or null if there is no result.
    *
    * <p>To update see {@link #execute(String, Handler)}.
    *
    * @param sql  The sql query to run.
    * @param replyHandler  The query result or the failure.
    */
   public void selectSingle(String sql, Handler<AsyncResult<Row>> replyHandler) {
-    getSQLConnection(conn -> selectSingle(conn, sql, closeAndHandleResult(conn, replyHandler)));
+    getSQLReadConnection(conn -> selectSingle(conn, sql, closeAndHandleResult(conn, replyHandler)));
```

Comment:
sql can be a writing SQL: `UPDATE foo SET jsonb='{}' where id='123' RETURNING jsonb`

Therefore this change cannot be made in this `selectSingle` method but requires a new `selectSingleRead` method.